tee-worker/core/rpc-client/src/direct_client.rs

```diff
@@ -207,6 +209,22 @@ impl DirectApi for DirectClient {
 	fn close(&self) -> Result<()> {
 		self.web_socket_control.close_connection()
 	}
+
+	fn get_next_nonce(&self, shard: &ShardIdentifier, account: &AccountId) -> Result<String> {
+		// let data = Request { shard, cyphertext: account.encode() };
+		let jsonrpc_call: String = RpcRequest::compose_jsonrpc_call(
+			"author_getNextNonce".to_owned(),
+			vec![shard.encode().to_base58(), account.to_hex()],
+		)
+		.unwrap();
+		debug!("[+] get_next_nonce jsonrpc_call: {}", jsonrpc_call);
+		// Send json rpc call to ws server.
+		let response_str = self.get(&jsonrpc_call)?;
+		debug!("[+] get_next_nonce response_str: {}", response_str);
+		let nonce_string = decode_from_rpc_response(&response_str)?;
```

Comment:
Are you sure this will work?

It seems to try to decode the `rpc_return_value.value` as string, but it's encoded from u32: `value: (nonce + pending_tx_count).encode(),`